xchange-core/src/main/java/org/knowm/xchange/BaseExchange.java

```diff
@@ -76,30 +76,26 @@ public void applySpecification(ExchangeSpecification exchangeSpecification) {
     if (this.exchangeSpecification.getMetaDataJsonFileOverride()
         != null) { // load the metadata from the file system
 
-      InputStream is = null;
-      try {
-        is = new FileInputStream(this.exchangeSpecification.getMetaDataJsonFileOverride());
+      try(InputStream is = new FileInputStream(this.exchangeSpecification.getMetaDataJsonFileOverride())) {
         loadExchangeMetaData(is);
-      } catch (FileNotFoundException e) {
+      } catch (IOException e) {
```

Comment:
I don't think that if the metadata file is malformed then only a warn in the logs is enough, I would prefer the exception to get out

Reply:
Do you mean to log the change the log level to error and log the error?

Reply:
This: log.error(e.getMessage(),e);

Reply:
no, I would prefer the IOException to get thrown out of this method into my application code

Reply:
when i am trying to throw the exception in the catch block, i get "Unhandled exception: IOException error". So i don't understand how else i can throw the exception? Do i need to add throws IOException on the method?

Reply:
ok, I didn't notice that applySpecification has no throws IOException

you have an example how to handle this at the end of the `applySpecification' method

```
   catch (IOException e) {
        throw new ExchangeException(e);
      }
```

Reply:
Thanks a lot for your response @walec51 ,although i have not understand how to do it? Do you want to write a TODO in order to add it later?

Reply:
just replace those two catches with this code:

```
 catch (IOException e) {
        throw new ExchangeException(e);
      }
```

you have to do a simple copy-paste, I don't know how to explain this simpler...

Reply:
xaxaxaxa. Sorry man, i thought it would throw the same error. I am making the push.